src/adapter/db/wishlist.gateway.ts

```diff
@@ -0,0 +1,68 @@
+import { EntityManager } from '@mikro-orm/core';
+import { EntityRepository } from '@mikro-orm/mysql';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { Inject, Injectable } from '@nestjs/common';
+import { WishlistNotFoundException } from 'src/application/common/error/exception/wishlist.exception';
+import { AddWishlistCommand } from 'src/application/port/in/wishlist/WishlistUseCase';
+import { WishlistDbCommandPort } from 'src/application/port/out/WishlistDbCommandPort';
+import { WishlistDbQueryPort } from 'src/application/port/out/WishlistDbQueryPort';
+import { Wishlist } from 'src/domain/wishlist';
+
+import { WishlistDbEntity } from './wishlist.entity';
+import { mapToWishlist } from './wishlist.mapper';
+
+@Injectable()
+export class WishlistGateway implements WishlistDbCommandPort, WishlistDbQueryPort {
+  constructor(
+    @InjectRepository(WishlistDbEntity)
+    private readonly wishlistRepository: EntityRepository<WishlistDbEntity>,
+    @Inject(EntityManager)
+    private readonly em: EntityManager,
+  ) {}
+
+  async getWishlistById(id: number): Promise<Wishlist> {
+    const wishlist = await this.wishlistRepository.findOne({ id });
+
+    if (!wishlist) {
+      throw new WishlistNotFoundException();
+    }
+
+    return mapToWishlist(wishlist);
+  }
+
+  async getUserWishlistByUserId(userId: number): Promise<Wishlist[]> {
+    const entities = await this.wishlistRepository.find(
+      { userId },
+      { orderBy: { createdAt: 'desc' } },
+    );
+    return entities.map(mapToWishlist);
+  }
+
+  async getByUserIdAndReceiverName(userId: number, receiverName: string): Promise<Wishlist[]> {
```

Comment:
이런식으로 네이밍해도 괜찮지 않나 싶네용
```suggestion
  async getUserWishlistsByReceiverName(userId: number, receiverName: string): Promise<Wishlist[]> {
```

Reply:
수정하였습니다!